src/taco-dependency-installer/installers/androidPackagesInstaller.ts

```diff
@@ -0,0 +1,296 @@
+﻿/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../../typings/dependencyInstallerInterfaces.d.ts" />
+/// <reference path="../../typings/Q.d.ts" />
+
+"use strict";
+
+import childProcess = require ("child_process");
+import fs = require ("fs");
+import os = require ("os");
+import path = require ("path");
+import Q = require ("q");
+
+import InstallerBase = require ("./installerBase");
+import installerProtocol = require ("../elevatedInstallerProtocol");
+import resources = require ("../resources/resourceManager");
+import tacoUtils = require ("taco-utils");
+
+import ILogger = installerProtocol.ILogger;
+import utilHelper = tacoUtils.UtilHelper;
+
+class AndroidPackagesInstaller extends InstallerBase {
+    private static ANDROID_HOME_NAME: string = "ANDROID_HOME";
+    private static PLATFORM_TOOLS_PKG: string = "platform-tools";
+    private static BUILD_TOOLS_PKG_PREFIX: string = "build-tools-";
+
+    private androidCommand: string;
+    private adbCommand: string;
+    private availablePackages: string[];
+
+    constructor(installerInfo: DependencyInstallerInterfaces.IInstallerData, softwareVersion: string, installTo: string, logger: ILogger, steps: DependencyInstallerInterfaces.IStepsDeclaration) {
+        super(installerInfo, softwareVersion, installTo, logger, steps, "androidPackages");
+    }
+
+    protected installWin32(): Q.Promise<any> {
+        return this.installDefault();
+    }
+
+    protected installDarwin(): Q.Promise<any> {
+        return this.installDefault();
+    }
+
+    private buildAndroidCommands(): void {
+        // Read the ANDROID_HOME from the process
+        var androidHomeValue: string = process.env[AndroidPackagesInstaller.ANDROID_HOME_NAME];
+
+        if (androidHomeValue) {
+            // ANDROID_HOME is set, use that to find the android executable
+            androidHomeValue = utilHelper.expandEnvironmentVariables(androidHomeValue);
+
+            // Build the android and adb commands and save them in members for reusability
+            var androidExecutable: string = os.platform() === "win32" ? "android.bat" : "android";
+            var adbExecutable: string = os.platform() === "win32" ? "adb.exe" : "adb";
```

Comment:
This pattern of constantly checking the platform in the code is making testing more difficult. @digeff had to mock the process object in his tests because of this. An easy way to fix this is to create an interface which contains all the platform specific details and to have two implementations, one for Windows and one for OS X. That way, if we add another platform later (say, Linux), we don't have to modify the code, but extend it (Open Closed Principle) See this for more info: http://www.objectmentor.com/resources/articles/ocp.pdf